### PR TITLE
Mohsen / feat: Add new key extraction

### PR DESCRIPTION
**Pull request by michio-rework**
Changes:

- added new translation key extraction method.
- updated the `localize` function and `Localize` component to be able to handle the new method.
- there is no breaking changes.
- the hero component and vertical-carousel in home page is refactored with this new approach.
- Please ignore the `Sonar Could` error / bug, since that part of the code will not be included in production. 

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: crowdin/messages.json [hunk 1]
```diff
@@ -1293,6 +1293,7 @@
   "1760558932": "Apple",
   "1764495771": "How can I close my account?",
   "1765454171": "Commission rate",
+  "1767169921": "Just testing some stuff",
```
Comment (balakrishna-binary) [inline crowdin/messages.json:1296]: should we include this? 😅

Comment (michio-rework) [reply crowdin/messages.json:1296]: No it's just there for show casing the new key extraction approach, I added a TODO comment in the actual file to remove this before merge :sweat_smile:

**Review by balakrishna-binary — APPROVED** [review]

**Review by niloo-fs — DISMISSED** [review]
LGTM, great job @michio-rework , thank u for your good idea about this :))

**Review by habib-deriv — APPROVED** [review]
LGTM

**Review by habib-deriv — APPROVED** [review]
LGTM

**Pull request merged**